### PR TITLE
Run pre-commit type check only for TypeScript

### PR DESCRIPTION
**Pull request by webfiltered**
Pros:
- Only runs type check if there is a `.ts`, `.tsx` or `.vue` file staged
- Is otherwise the same as current `npm typecheck` (inc. `tsc-strict`)

Cons:
- Changing other files, e.g. `tsconfig.json` without any .ts files staged, type checking will not run

### CODE REVIEW

**Comment by webfiltered** [comment]
This one is just an option, because I already did the work on the other side.  There's nothing relying on this - feel free to close!

File: lint-staged.config.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+export default {
+  // '*.css': stagedFiles => `prettier --write ${stagedFiles.join(' ')}`,
```
Comment (huchenlei) [inline lint-staged.config.js]: nit: remove commented code.

File: lint-staged.config.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    ...formatFiles(stagedFiles),
+    'tsc --noEmit',
+    'tsc-strict'
+  ]
+}
+
+function formatFiles(fileNames) {
+  return [
+    `prettier --write ${fileNames.join(' ')}`
+    // `eslint --fix ${fileNames.join(' ')}`,
```
Comment (huchenlei) [inline lint-staged.config.js]: nit: remove commented code.

**Review by huchenlei — APPROVED** [review]
LGTM with nits.

File: lint-staged.config.js (hunk 2 above)
Comment (webfiltered) [reply lint-staged.config.js]: Done!  Left those in case you wanted to uncomment them.

**Pull request merged**